sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/codegen/UnsafeWriter.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql.catalyst.expressions.codegen;
+
+import org.apache.spark.sql.types.Decimal;
+import org.apache.spark.unsafe.types.CalendarInterval;
+import org.apache.spark.unsafe.types.UTF8String;
+
+/**
+ * Base class for writing Unsafe* structures.
+ */
+public abstract class UnsafeWriter {
+  public abstract void setNullByte(int ordinal);
```

Comment:
This looks pretty weird. At the first glance I'm wondering why we don't have `setBoolean/Float/Double`, then I realized we don't need to, because we just need a way to set null for 1/2/4/8 bytes.

maybe it's better to name them `setNull1/2/4/8Bytes`, and ask the `UnsafeArrayWriter` to follow

Reply:
See my previous discussion with @mgaido91. I am fine either way, I can also add the missing methods an be done with it, that will just make the interpreted code path a bit messier.

Reply:
I feel `setNull1/2/4/8Bytes` is better. It's also easy to codegen, just `setNull${dt.defaultSize}Bytes`.

Reply:
Actually this was my mistake... I thought `Platform.setInt(0)` is different from `Platform.setFloat(0.0f)`, and that's why I introduced a `setNull` method for each primitive type.